tests/density.js

```diff
@@ -2,22 +2,121 @@ import http from 'k6/http';
 import { check } from 'k6';
 import { Kubernetes } from 'k6/x/kubernetes';
 import * as deploy from "./common/common.js";
-import { Gauge, Counter } from 'k6/metrics';
+import { Gauge } from 'k6/metrics';
 import exec from 'k6/execution';
 import { sleep } from 'k6';
 
-export let options = {
-  stages: [
-    { target: 20, duration: '10s' },
-    { target: 0, duration: '10s' }
-  ],
-  setupTimeout: '300s',
-  noConnectionReuse: true
-};
-
+const testName = "density";
+const route = ""
+const replicas = 1;
+const namespace = `${__ENV.NAMESPACE}` != "undefined" ? `${__ENV.NAMESPACE}` : "default";
+const executor = `${__ENV.EXECUTOR}` != "undefined" ? `${__ENV.EXECUTOR}` : "containerd-shim-spin";
+const repo = `${__ENV.REPO}` != "undefined" ? `${__ENV.REPO}` : "rg.fr-par.scw.cloud/dlancshire-public/template-app-";
+const tag = `${__ENV.TAG}` != "undefined" ? `${__ENV.TAG}` : "latest";
 const timeToDeployTenApps = new Gauge('time_to_deploy_ten_apps');
 const deployedApps = new Gauge('deployed_apps');
-const waitAppsReadyTimeout = 300;
+const maxScenarioDurationSecs = `${__ENV.MAX_SCENARIO_DURATION_SECS}` != "undefined" ? `${__ENV.MAX_SCENARIO_DURATION_SECS}` : 180;
+const restIntervalSecs = `${__ENV.REST_INTERVAL_SECONDS}` != "undefined" ? `${__ENV.REST_INTERVAL_SECONDS}` : 10;
+
+// Should match the number of apps deployed by the last stage
+// and be divisible by the batch size
+// TODO: make this configurable
+const totalApps = 50;
+
+export let options = {
+  tags: {
+    test: testName,
+  },
+  noConnectionReuse: true,
+  discardResponseBodies: true,
```

Comment:
Whoops, this is the cause for the failed checks re: res.body.  Per https://grafana.com/docs/k6/latest/javascript-api/k6-http/params/, when this is set, k6 returns `null` for the body.

Reply:
Good catch thank you!